### ISSUE TITLE
CALLSTACK - [$1000] Cursor moves for every character clicked after max character limit is reached

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18925]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Open newDot on mobile web Chrome
2. Go to workspace > General settings
3. Put a long string in the "Name" input field until the max character limit is reached
4. Go to the last character and make sure you can't add any more characters
5. Click on the middle of the long string to put the cursor at the middle of the long Name
6. Click on any character on your keyboard

## Expected Result:
Nothing should happen

## Actual Result:
Cursor moves for every character clicked (but no characters are added to the name string)
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [x] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.13.0
**Reproducible in staging?:** needs reproduction
**Reproducible in production?:** needs reproduction
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/92cb6493-49b5-433c-a6d2-92c1513a9a8e

**Expensify/Expensify Issue URL:**
**Issue reported by:** @nathan-mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683883480593569

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012596222bed540b4b</li>
        <li>Upwork Job ID: 1658866428189360128</li>
        <li>Last Price Increase: 2023-06-07</li>
    </ul>
</details>

**Comment by maddylewis** [comment]
asking about this one here - https://expensify.slack.com/archives/C049HHMV9SM/p1684289212745439?thread_ts=1683883480.593569&cid=C049HHMV9SM

**Comment by maddylewis** [comment]
i can repro this - adding external label for review/proposals.

**Comment by mollfpr** [comment]
Thanks @payal-lathidadiya. I just tried your solution, but the cursor is still moving forward.

<details>
<summary>video</summary>

https://github.com/Expensify/App/assets/25520267/61a223f2-3780-4808-9ddc-8fd726165604


</details>

<details>
<summary>diff</summary>

```diff
diff --git a/src/components/TextInput/BaseTextInput.js b/src/components/TextInput/BaseTextInput.js
index 1a6eec10fb..0bc4ff42db 100644
--- a/src/components/TextInput/BaseTextInput.js
+++ b/src/components/TextInput/BaseTextInput.js
@@ -144,6 +144,10 @@ class BaseTextInput extends Component {
      * @memberof BaseTextInput
      */
     setValue(value) {
+        if (value.length >= this.props.maxLength) {
+            return;
+        }
+
         if (this.props.onInputChange) {
             this.props.onInputChange(value);
         }
```

</details>

**Comment by payal-lathidadiya** [comment]
@mollfpr thanks for reviewing my proposal and checking this.

This proposed change did fixed this behavior for me.

https://github.com/Expensify/App/assets/130229379/e4cd42be-9bbc-4468-8cd8-5f2d1b047af4




which android and chrome version you've?

**Comment by mollfpr** [comment]
No new proposal.

**Comment by maddylewis** [comment]
@srikarparsi @mollfpr - can either of you provide an update on the status of this one? have we decided on a proposed fix? lmk - thanks!

**Comment by mollfpr** [comment]
@maddylewis No accepted proposal yet. The [result](https://github.com/Expensify/App/issues/18925#issuecomment-1556601642) from @payal-lathidadiya proposal is not meet the expected result.

**Comment by maddylewis** [comment]
https://expensify.slack.com/archives/C03UK30EA1Z/p1685639275960899

**Comment by maddylewis** [comment]
Tomek said someone would likely pick this up today/tomorrow - https://expensify.slack.com/archives/C03UK30EA1Z/p1685739337368829?thread_ts=1685639275.960899&cid=C03UK30EA1Z

**Comment by srikarparsi** [comment]
Seems like it's a problem with the browser's default behavior: https://github.com/necolas/react-native-web/issues/2526. I'm not sure if there is a straightforward workaround for this or if this is worth resolving. @maddylewis what do you think?

**Comment by maddylewis** [comment]
i don't think this is worth fix personally. it doesn't seem like something that anyone would run into very often.

**Comment by payal-lathidadiya** [comment]
I did detailed investigation on this issue to find root cause and possible fix or work around.

The issue @srikarparsi mentioned above https://github.com/necolas/react-native-web/issues/2526 is the one I had raised with library maintainer.

Since there wasn't any resolution, I was investing if there can be any work around to have consistent UX across platform.

**Comment by payal-lathidadiya** [comment]
I did detailed investigation on this issue to find root cause and possible fix or work around.

The issue @srikarparsi mentioned above https://github.com/necolas/react-native-web/issues/2526 is the one I had raised with library maintainer.

Since there wasn't any resolution, I was investing if there can be any work around to have consistent UX across platform.

I feel sad that I couldn't help on viable resolution though considerable effort.

I'm also aligned with @maddylewis to live with this for now as it isn't much often and impacting functioning of application.
